src/driver/src/DefaultDriver.cpp

```diff
@@ -55,6 +66,103 @@ YAML::Node loadConfig(const std::string& source,
     }
 }
 
+YAML::Node parseExternal(const YAML::Node& external,
+                         YamlParameters& params,
+                         const fs::path& phaseConfig) {
+    int nodeSize = 1;
+
+    fs::path path(external["Path"].as<std::string>());
+    path = fs::absolute(phaseConfig / path);
+
+    if (!fs::is_regular_file(path)) {
+        auto os = std::ostringstream();
+        os << "Invalid path to external PhaseConfig: " << path
+           << ". Please ensure your workload file is placed in 'workloads/[subdirectory]/' and the "
+              "'Path' parameter is relative to the 'phase_configs/' directory";
+        throw InvalidConfigurationException(os.str());
+    }
+
+    auto replacement = loadConfig(path.string());
+
+    if (external["Parameters"]) {
+        nodeSize++;
+        auto newParams = external["Parameters"].as<YamlParameters>();
+        params.insert(newParams.begin(), newParams.end());
+    }
+
+    if (external["Key"]) {
+        nodeSize++;
+        const auto key = external["Key"].as<std::string>();
+        if (!replacement[key]) {
+            auto os = std::ostringstream();
+            os << "Could not find top-level key: " << key
+               << " in phase config YAML file: " << path.string();
+            throw InvalidConfigurationException(os.str());
+        }
+        replacement = replacement[key];
+    }
+
+    if (external.size() != nodeSize) {
+        throw InvalidConfigurationException(
+            "Invalid keys for 'External'. Please set 'Path' and if any, 'Parameters");
```

Comment:
also print the `external` `Node` - you can use `YAML::Emitter` for this - something like the following:

```c++
auto toString(YAML::Node n) {
  YAML::Emitter e;
  e << n;
  return e.str();
}
```

this would also be useful in other places where this code throws on bad yaml - print the full yaml that's in scope wherever possible.

Reply:
Done. I used `YAML::Dump()` instead to save a `.str()` call.